File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s)!=len(t):
            return False
        d1={}
        d2={}
        for ch in s:
            d1[ch]=d1.get(ch,0)+1
        for a in t:
            d2[a]=d2.get(a,0)+1
        return d1==d2
    

File: 0242-valid-anagram/0242-valid-anagram.c
#include <string.h>
bool isAnagram(char* s, char* t) {
    int hash1[26]={0};
    int hash2[26]={0};
    for (int i=0;i<strlen(s);i++){
        hash1[s[i]-'a']++;
    }
    for (int i=0;i<strlen(t);i++){
        hash2[t[i]-'a']++;
    }
    for (int i=0;i<26;i++){
        if (hash1[i]!=hash2[i]){
            return false;
        }
    }
    return true;
}